### PR TITLE
Make unit tests work with ZConfig >= 3.4.0, refresh supported Python versions

### LINKED ISSUE #5: Tests are broken
See https://travis-ci.com/github/zopefoundation/zope.app.server/builds/194727326

```
Test-module import failures:

Module: zope.app.server.tests.test_accesslog

Traceback (most recent call last):
  File "/home/travis/build/zopefoundation/zope.app.server/src/zope/app/server/tests/test_accesslog.py", line 22, in <module>
    from ZConfig.components.logger.tests.test_logger import LoggingTestBase
ImportError: cannot import name 'LoggingTestBase'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/travis/build/zopefoundation/zope.app.server/src/zope/app/server/tests/test_accesslog.py", line 24, in <module>
    from ZConfig.components.logger.tests.test_logger import LoggingTestHelper
ImportError: cannot import name 'LoggingTestHelper'
```

It seems because we use code from `ZConfig` tests which has been moved.

### PR DESCRIPTION
**Pull request by mgedmin**
Fixes #5.

### CODE REVIEW

**Comment by mgedmin** [comment]
Python 3.4 test failed on Travis, and I'm not going to investigate -- it's EOL, long past time to drop support.

PyPy test needs the usual adjustment due to an old OpenSSL in the xenial pypy images (switch to bionic or focal, use pypy2 instead of pypy).

File: src/zope/app/server/tests/test_accesslog.py [hunk 1]
```diff
@@ -19,12 +19,21 @@
 from ZConfig.components.logger import loghandler
 
 try:
+    # ZConfig < 2.9.2
```
Comment (icemac) [inline src/zope/app/server/tests/test_accesslog.py:22]: Should we actually still support this 8 year old version?

**Review by icemac — APPROVED** [review]
LGTM with one question.

File: src/zope/app/server/tests/test_accesslog.py (hunk 1 above)
Comment (mgedmin) [reply src/zope/app/server/tests/test_accesslog.py:22]: Probably not, but I was too lazy to remove it.

**Pull request merged**